how-we-work/how-to-upgrade-dependencies.md

```diff
@@ -0,0 +1,75 @@
+
+For an overview of current dependency issues we have in cellxgene Explorer and cellxgene Portal, you can check the two corresponding projects in [Snyk](https://app.snyk.io/org/cellxgene). Note that you should take the findings with a grain of salt as some findings may not have remediation and therefore may continue to exist after everything else has been upgraded. It's a good place to double check your work though.
+
+## Upgrading cellxgene Explorer
+
+### Upgrading cellxgene Explorer frontend
+
+There are two commands to run to figure out which npm packages to upgrade:
+
+`npm audit` -- this command generates a report of known vulnerabilities in dependencies described in `cellxgene/client/package.json`.
+
+`npm outdated` -- this command generates a report of all dependencies in `cellxgene/client/package.json` and whether there are newer versions available.
+
+Run these two commands and upgrade packages to the latest available. Please also make any necessary changes in the client code to ensure that that the frontend still works as expected and is not using outdated references. Finally, run `npm i` to install the packages and update the corresponding `package-lock.json`.
+
+[Here](https://github.com/chanzuckerberg/cellxgene/pull/2167/files) is a sample PR that upgraded several frontend npm dependencies.
+
+### Upgrading cellxgene Explorer CZI-hosted backend
+
+The pip package dependency files for the CZI-hosted backend for cellxgene Explorer can be found at `cellxgene/backend/czi-hosted/requirements.txt` as well as `single-cell-infra/terraform/modules/hosted-cellxgene/common/customize/requirements.txt`. These two files should be in sync except that the file in `single-cell-infra` should pin all packages to a single version and the file in `cellxgene` should use the `>=` sign for the same version numbers. This is so that when we are testing locally, we can easily find out if a new release of a package is causing compatibility issues. We do want to pin versions in the `single-cell-infra` repo because these are the versions installed during deployment and we don't want an unexpected incompatibility due to a new package release to cause a production deployment to fail.
+
+To perform the upgrades you can make use of `pip-upgrader` to figure out which packages have recent updates:
+
+1. Install [pip-upgrader](https://pypi.org/project/pip-upgrader/): `pip install pip-upgrader`.
+
+2. Upgrade the packages based on the report and verify that the application still works (please fully deploy to a `dev` or `rdev` environment to double check everything).
+
+You can also make use of the [Snyk reports](https://app.snyk.io/org/cellxgene) to figure out which specific packages have security vulnerabilities as well as the [`safety` package](#Upgrade-cellxgene-Explorer-desktop-backend).
```

Comment:
Note that `safety` evaluates the *currently installed* packages.  To use it you have to create a clean virtual env, and install from the requirements file -- then run safety.

Reply:
Added prose around this.